ilastik/applets/counting/countingGuiBoxesInterface.py

```diff
@@ -970,41 +973,39 @@ def _getNextBoxColor(self):
         return color
 
     def saveBoxesToCSV(self, filename):
-        import os, csv
-
-        b, ext = os.path.splitext(str(filename))
-        assert ext == ".txt", "wrong filename or extension %s" % str(filename)
         try:
-            with open(filename, "wb") as fh:
+            with open(filename, "w", newline="") as csvfile:
+                writer = csv.writer(csvfile, quoting=csv.QUOTE_MINIMAL)
+                writer.writerow(["ID", "StartX", "StartY", "StopX", "StopY", "Count", "Average density", "Std density"])
+
+                for row_id, box in enumerate(self._currentBoxesList):
+                    start = [str(p) for p in box.getRectItem().topLeftDataPos()]
+                    stop = [str(p + 1) for p in box.getRectItem().bottomRightDataPos()]
+                    region = box.getSubRegion()
+                    count = format(np.sum(region), "0.8g")
```

Comment:
just out of curiosity, why `format` and not `f"{np.sum(region):0.8g}"`?

Reply:
I had f-strings before, but `format` is marginally better because all these variables should be just numbers in a string form, not arbitrary strings. Besides, I didn't knew about `format` before :)